src/modal/test/modal.spec.js

```diff
@@ -27,60 +27,91 @@ describe('$modal', function () {
-        this.message = function() {
-          return 'Expected "' + angular.mock.dump(resolved) + '" to be resolved with "' + value + '".';
-        };
-        this.actual.then(function(result){
-          resolved = result;
-        });
-        $rootScope.$digest();
+              if (result !== expected) {
+                result.message = 'Expected "' + angular.mock.dump(result) + '" to be resolved with "' + expected + '".';
+              }
+            });
 
-        return resolved === value;
-      },
+            $rootScope.$digest();
 
-      toBeRejectedWith: function(value) {
-        var rejected;
-        this.message = function() {
-          return 'Expected "' + angular.mock.dump(rejected) + '" to be rejected with "' + value + '".';
+            return {pass: true};
+          }
         };
-        this.actual.then(angular.noop, function(reason){
-          rejected = reason;
-        });
-        $rootScope.$digest();
-
-        return rejected === value;
       },
+      toBeRejectedWith: function(util, customEqualityTesters) {
+        return {
+          compare: function(promise, expected) {
+            var result = {};
 
-      toHaveModalOpenWithContent: function(content, selector) {
+            promise.then(function() {
 
-        var contentToCompare, modalDomEls = this.actual.find('body > div.modal > div.modal-dialog > div.modal-content');
+            }, function(result) {
+              expect(result).toEqual(expected);
 
-        this.message = function() {
-          return '"Expected "' + angular.mock.dump(modalDomEls) + '" to be open with "' + content + '".';
-        };
+              if (result !== expected) {
+                result.message = 'Expected "' + angular.mock.dump(result) + '" to be rejected with "' + expected + '".';
+              }
+            });
 
-        contentToCompare = selector ? modalDomEls.find(selector) : modalDomEls;
-        return modalDomEls.css('display') === 'block' &&  contentToCompare.html() == content;
+            $rootScope.$digest();
+
+            return {pass: true};
+          }
+        };
       },
+      toHaveModalOpenWithContent: function(util, customEqualityTesters) {
+        return {
+          compare: function(actual, content, selector) {
+            var contentToCompare, modalDomEls = actual.find('body > div.modal > div.modal-dialog > div.modal-content');
 
-      toHaveModalsOpen: function(noOfModals) {
+            contentToCompare = selector ? modalDomEls.find(selector) : modalDomEls;
 
-        var modalDomEls = this.actual.find('body > div.modal');
-        return modalDomEls.length === noOfModals;
-      },
+            var result = {
+              pass: modalDomEls.css('display') === 'block' && contentToCompare.html() === content
+            };
 
-      toHaveBackdrop: function() {
+            if (!result.pass) {
+              result.message = '"Expected "' + angular.mock.dump(modalDomEls) + '" to be open with "' + content + '".';
```

Comment:
There needs to a message for when the pass is true because the message is displayed when the condition is used with `.not`. Like if `.not.toHaveModalOpenWithContent` fails.